### DISCUSSION TITLE
iDeal showing in payment sheed On IOS

### DISCUSSION THREAD

**Discussion by cm-creativecentral** [flutter-stripe/flutter_stripe#1999] [Q&A]
**Describe the bug** On iOS devices, the payment sheet only displays the Card payment option, whereas on Android devices, both Card and iDEAL payment options are available. Users on iOS are unable to select iDEAL as a payment method, even though it should be supported.

iDEAL is a widely used payment method, and its absence on iOS could inconvenience users who prefer it over other payment options. This issue may affect transaction completion rates on iOS devices.

**Comment by cm-creativecentral** [comment]
![WhatsApp Image 2024-11-10 at 15 02 44_3c4f3e84](https://github.com/user-attachments/assets/eaed0c31-4d68-46f4-89fa-3ef85aa101c2)

**Comment by remonh87** [comment]
Ideal works in the payment sheet do you charge the customer in euros? Ideal only works in euros . I just paid with ideal this morning 😉 using our sdk

**Comment by cm-creativecentral** [comment]
> Ideal works in the payment sheet do you charge the customer in euros? Ideal only works in euros . I just paid with ideal this morning 😉 using our sdk

i am working with euros. and its working on Andriod and not working on IOS.

i am using 'flutter_stripe: ^11.2.0'

![WhatsApp Image 2024-11-27 at 23 38 39_f14b683f](https://github.com/user-attachments/assets/b6776b3c-865a-4218-bb54-3e4db77b5e82)
![WhatsApp Image 2024-11-27 at 23 38 26_819e60c6](https://github.com/user-attachments/assets/723a50d6-5e95-4061-9b3f-d6859eeea8bc)

**Comment by remonh87** [comment]
Ah you need to provide a valid return_url as parameter. Stripe will execute the return url after the payment succeeds in order to return to the app

**Comment by cm-creativecentral** [comment]
> Ah you need to provide a valid return_url as parameter. Stripe will execute the return url after the payment succeeds in order to return to the app

But on android works without any return URL. Is it possible to share integration code snippet of that

**Comment by remonh87** [comment]
https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/regional_payment_methods/ideal_screen.dart

and 

https://docs.flutter.dev/ui/navigation/deep-linking

**Reply by cm-creativecentral** [reply]
> https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/regional_payment_methods/ideal_screen.dart
> 
> and
> 
> https://docs.flutter.dev/ui/navigation/deep-linking

![image](https://github.com/user-attachments/assets/510875a9-d2a3-40de-a8ca-c2d43c42ce00)
after passing return url still it was taking defalu stripesdk:// url on andriod but on ios it was working correctly.
i have checked deeplinking also working file on andriod but return url is not setting properly

![image](https://github.com/user-attachments/assets/0394f5fd-372c-47cc-9054-82f7eb1c58ea)
